app/src/main/java/com/pluu/webtoon/data/daum/DaumWeekApi.kt

```diff
@@ -0,0 +1,97 @@
+package com.pluu.webtoon.data.daum
+
+import com.pluu.kotlin.asSequence
+import com.pluu.kotlin.isNotEmpty
+import com.pluu.webtoon.common.Const
+import com.pluu.webtoon.data.IRequest
+import com.pluu.webtoon.data.REQUEST_METHOD
+import com.pluu.webtoon.data.WeeklyRequest
+import com.pluu.webtoon.data.impl.AbstractWeekApi
+import com.pluu.webtoon.di.NetworkUseCase
+import com.pluu.webtoon.item.Result
+import com.pluu.webtoon.item.Status
+import com.pluu.webtoon.item.ToonInfo
+import com.pluu.webtoon.utils.safeAPi
+import org.json.JSONArray
+import org.json.JSONObject
+import java.text.SimpleDateFormat
+import java.util.*
+
+/**
+ * 다음 웹툰 Week Api
+ * Created by pluu on 2017-04-20.
+ */
+class DaumWeekApi(
+    networkUseCase: NetworkUseCase
+) : AbstractWeekApi(networkUseCase, DaumWeekApi.TITLE) {
+
+    private val URL_VALUE = arrayOf("mon", "tue", "wed", "thu", "fri", "sat", "sun")
+
+    override fun parseMain(param: WeeklyRequest): Result<List<ToonInfo>> {
+        ///////////////////////////////////////////////////////////////////////////
+        // API
+        ///////////////////////////////////////////////////////////////////////////
+
+        val apiResult = safeAPi(requestApi(createApi(param))) { response ->
+            JSONObject(response)
+        }
+
+        val responseData = when (apiResult) {
+            is Result.Success -> apiResult.data
+            is Result.Error -> return Result.Error(apiResult.exception)
+        }
+
+        ///////////////////////////////////////////////////////////////////////////
+        // Parse Data
+        ///////////////////////////////////////////////////////////////////////////
+
+        val array: JSONArray? = responseData
+            .optJSONObject("data")
+            .optJSONArray("webtoons")
+
+        if (array?.isNotEmpty() != true) {
+            return Result.Success(emptyList())
+        }
+
+        val today = SimpleDateFormat("yyyyMMdd", Locale.getDefault()).format(Date())
+
+        val list = array.asSequence().map { obj ->
+            val lastObj = obj.optJSONObject("latestWebtoonEpisode")
+            val date = lastObj.optString("dateCreated")
+
+            ToonInfo(
+                id = obj.optString("nickname"),
+                title = obj.optString("title"),
+                image = lastObj.optJSONObject("thumbnailImage").optString("url"),
+                writer = obj.optJSONObject("cartoon")
+                    .optJSONArray("artists")
+                    .optJSONObject(0)
+                    .optString("name"),
+                rate = Const.getRateNameByRate(obj.optString("averageScore")).takeIf { it != "0.0" }.orEmpty(),
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/Pluu/WebToon/blob/e6d7b5dbb11d2d4fb917c63fbb0187d6361e5aea/app/src/main/java/com/pluu/webtoon/data/daum/DaumWeekApi.kt#L70"></span>Exceeded max line length (100) (cannot be auto-corrected)</td>
    </tr>
  </tbody>
</table>